### PR TITLE
PR to apply libmobilecion changes from v1.2.2 to master

### PR DESCRIPTION
**Pull request by carybakker**
### Motivation

A number of changes were made to the libmobilecoin folder in the v1.2.2 release branch that were not applied to master.  This change is to apply any v1.2.2 libmobilecoin changes to master without clobbering any parallel updates made to the same files in the master branch.

In the commit comments, the individual commits for this PR document the source commit hash from the v1.2.2 branch for the changes being applied.

One change was a minor bug fix to `libmobilecoin/include/transaction.h` where a missing comma in the parameter list was added for `mc_memo_gift_code_funding_memo_create()`

For reference, these are the 10 commits for changes that existed in v1.2.2 and not in master for the libmobilecoin subfolder:
 * b3eb1e384fd4dfe5b60fb263c17c8009fd6666ea
 * 196ca2eb14a3c4f377e8e471f80070dafde40f4a
 * c27b390b535b96bf1c1b941d8da9502e758dec67
 * 31f140f865ceb52e48ed956c5ef335e319b641b2
 * 51fdac4085cd5fb6946a61797193bbe94e7d9a27
 * c3963f3af62ba820675401a30b5856cee2b5de2c
 * f44789181c2865ec82f766fd0b5655814e29feac
 * 65c3076b2c45a06dbef5f213e3983897ecf9403f
 * e4f41d189d130a69c1065715c3e47ecabfd7abe2
 * 73ee43066f441b18c62de4bfba9f2acea1d79628

Of those changes, 6 were Cargo.toml changes for version numbers.

The commit hashes for the remaining material changes are:
 * c27b390b535b96bf1c1b941d8da9502e758dec67
 * c3963f3af62ba820675401a30b5856cee2b5de2c
 * e4f41d189d130a69c1065715c3e47ecabfd7abe2
 * 73ee43066f441b18c62de4bfba9f2acea1d79628

### CODE REVIEW

**Review by the-real-adammork — APPROVED** [review]
Looks good to me

File: Cargo.toml [hunk 1]
```diff
@@ -191,6 +191,7 @@ lto = true
 [profile.mobile-release]
 inherits = "mobile"
 codegen-units = 1
+lto = false
```
Comment (garbageslam) [inline Cargo.toml:194]: any reason why you guys wanted to do this? I thought when we turned this on it made the binary considerably smaller

**Review by garbageslam — APPROVED** [review]
LGTM

**Comment by the-real-adammork** [comment]
We're seeing these linker errors when compiling the static library in the swift project, wonder if this can be resolved with a change in the rust code ?


```
Ld /Users/adam/Library/Developer/Xcode/DerivedData/Example-amigwsjyfbmoeibzmttqdrgsyjiw/Build/Products/Debug-iphonesimulator/LibMobileCoin/LibMobileCoin.framework/LibMobileCoin normal (in target 'LibMobileCoin' from project 'Pods')
    cd /Users/Shared/MobileCoin-Swift-MobileDev/ExampleHTTP/Pods
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -target x86_64-apple-ios10.0-simulator -dynamiclib -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator15.5.sdk -L/Users/adam/Library/Developer/Xcode/DerivedData/Example-amigwsjyfbmoeibzmttqdrgsyjiw/Build/Products/Debug-iphonesimulator/LibMobileCoin -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphonesimulator -L/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator15.5.sdk/usr/lib/swift -F/Users/adam/Library/Developer/Xcode/DerivedData/Example-amigwsjyfbmoeibzmttqdrgsyjiw/Build/Products/Debug-iphonesimulator/LibMobileCoin -F/Users/adam/Library/Developer/Xcode/DerivedData/Example-amigwsjyfbmoeibzmttqdrgsyjiw/Build/Products/Debug-iphonesimulator/SwiftProtobuf -filelist /Users/adam/Library/Developer/Xcode/DerivedData/Example-amigwsjyfbmoeibzmttqdrgsyjiw/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/LibMobileCoin.build/Objects-normal/x86_64/LibMobileCoin.LinkFileList -install_name @rpath/LibMobileCoin.framework/LibMobileCoin -Xlinker -rpath -Xlinker /usr/lib/swift -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -dead_strip -Xlinker -object_path_lto -Xlinker /Users/adam/Library/Developer/Xcode/DerivedData/Example-amigwsjyfbmoeibzmttqdrgsyjiw/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/LibMobileCoin.build/Objects-normal/x86_64/LibMobileCoin_lto.o -Xlinker -export_dynamic -Xlinker -objc_abi_version -Xlinker 2 -Xlinker -debug_variant -fobjc-arc -fobjc-link-runtime -fprofile-instr-generate -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphonesimulator -L/usr/lib/swift -Xlinker -add_ast_path -Xlinker /Users/adam/Library/Developer/Xcode/DerivedData/Example-amigwsjyfbmoeibzmttqdrgsyjiw/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/LibMobileCoin.build/Objects-normal/x86_64/LibMobileCoin.swiftmodule /Users/Shared/MobileCoin-Swift-MobileDev/ExampleHTTP/Pods/../../Vendor/libmobilecoin-ios-artifacts/Artifacts/target/x86_64-apple-ios/release/libmobilecoin.a -u _mc_string_free -framework SwiftProtobuf -framework Foundation -Xlinker -no_adhoc_codesign -compatibility_version 1 -current_version 1 -Xlinker -dependency_info -Xlinker /Users/adam/Library/Developer/Xcode/DerivedData/Example-amigwsjyfbmoeibzmttqdrgsyjiw/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/LibMobileCoin.build/Objects-normal/x86_64/LibMobileCoin_dependency_info.dat -o /Users/adam/Library/Developer/Xcode/DerivedData/Example-amigwsjyfbmoeibzmttqdrgsyjiw/Build/Products/Debug-iphonesimulator/LibMobileCoin/LibMobileCoin.framework/LibMobileCoin

Undefined symbols for architecture x86_64:
  "_deflateInit_", referenced from:
      _ssl_populate_transform in libmobilecoin.a(ssl_tls.c.o)
  "_inflateInit_", referenced from:
      _ssl_populate_transform in libmobilecoin.a(ssl_tls.c.o)
  "_deflateEnd", referenced from:
      _mbedtls_ssl_transform_free in libmobilecoin.a(ssl_msg.c.o)
  "_inflateEnd", referenced from:
      _mbedtls_ssl_transform_free in libmobilecoin.a(ssl_msg.c.o)
  "_inflate", referenced from:
      _mbedtls_ssl_read_record in libmobilecoin.a(ssl_msg.c.o)
  "_deflate", referenced from:
      _mbedtls_ssl_write_record in libmobilecoin.a(ssl_msg.c.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by the-real-adammork** [comment]
@garbageslam I was trying to fix bitcode support and saw this on the Apple Developer docs:

> Undefined symbols when using assembly code or inline assembly with link-time optimization (LTO).
> When using LTO, symbols in assembly code may be obfuscated inconsistently.
> 
> Workaround: Avoid LTO or rewrite the assembly code to avoid symbol references.

[https://developer.apple.com/library/archive/technotes/tn2432/_index.html](https://developer.apple.com/library/archive/technotes/tn2432/_index.html)

**Review by the-real-adammork — CHANGES_REQUESTED** [review]
see linker comment

**Comment by the-real-adammork** [comment]
@carybakker made a PR to merge a change into your branch - https://github.com/carybakker/mobilecoin/pull/1 - its needed in master but makes more sense to merge in ontop of the 1.2.x stuff which is what your PR here is doing.

**Review by the-real-adammork — APPROVED** [review]
re-ran locally and everything is building correctly. phil verified unit/integration tests on Swift. this is good to merge

**Pull request merged**